ivy/func_wrapper.py

```diff
@@ -771,6 +772,22 @@ def _infer_device(*args, device=None, **kwargs):
     return _infer_device
 
 
+def auto_shift_device(fn: Callable) -> Callable:
+    def _auto_shift_device(*args, **kwargs):
+        if ivy.functional.device.soft_device_placement and ivy.get_array_mode():
+            device = ivy.default_device()
+            args = (
+                arg.to_device(device) if type(arg) == ivy.Array else arg for arg in args
+            )
+            kwargs = {
+                k: v.to_device(device) if type(v) == ivy.Array else v
+                for k, v in kwargs.items()
+            }
+        return fn(*args, **kwargs)
+
+    return _auto_shift_device
```

Comment:
should add `_auto_shift_device.auto_shift_device = True` above this

Reply:
Thanks Sherry, this single line of code saved a lot of debugging hours for me😂